src/core/thread.d

```diff
@@ -2574,12 +2511,10 @@ extern (C) void thread_suspendAll() nothrow
             {
                 Thread.remove(t);
             }
-            else if (t.m_isInCriticalRegion)
+            else if (atomicLoad(t.m_isInCriticalRegion))
```

Comment:
I need sequential consistency here, because I must not send a suspension signal as soon as another thread sets it's `m_isInCriticalRegion` flag.